src/qibolab/couplers.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass, field
+from typing import List, Optional, Tuple, Union
+
+from qibolab.channels import Channel
+from qibolab.native import TwoQubitNatives
+from qibolab.qubits import Qubit, QubitId
+
+CouplerId = Union[str, int]
+"""Type for Coupler names."""
+
+
+@dataclass
+class Coupler:
+    """Representation of a physical coupler.
+
+    Coupler objects are instantiated by :class:`qibolab.platforms.platform.Platform`
+    but they are passed to instrument designs in order to play pulses.
+
+    Args:
+        name (int, str): Qubit number or name.
+
+        flux (:class:`qibolab.platforms.utils.Channel`): Channel used to
+            send flux pulses to the qubit.
+        Other characterization parameters for the coupler, loaded from the runcard.
+    """
+
+    name: CouplerId
+
+    # TODO: I think is not needed
+    frequency: int = 0
+
+    sweetspot: float = 0
+
+    flux: Optional[Channel] = None
+    qubits: Optional[List["Qubit"]] = field(default_factory=dict)
+
+    def __post_init__(self):
+        # register qubit in ``flux`` channel so that we can access
+        # ``sweetspot`` and ``filters`` at the channel level
+        if self.flux:
+            self.flux.coupler = self
```

Comment:
This is probably copied from the `Qubit` class but I think it is not needed anymore (it was leftover from some previous code). In fact it should also be removed from the `Qubit`. It is not very good practice anyway, as it is dynamically adding an attribute to a `Channel` which does not exist in the `Channel` class definition.